### PR TITLE
Add iperf client example

### LINKED ISSUE #255: iperf example fails to build when compiled as a client
Complains that xstr is not defined.

### PR DESCRIPTION
**Pull request by peterharperuk**
Requires you to specify the server ip address when running cmake,
e.g. -DIPERF_SERVER_IP=1.2.3.4

Fixes #255

### CODE REVIEW

**Comment by lurch** [comment]
Perhaps the top-level `README.md` also wants updating to match?

**Comment by peterharperuk** [comment]
> Perhaps the top-level `README.md` also wants updating to match?

ok - tweak the wording

File: pico_w/iperf/picow_iperf.c [hunk 1]
```diff
@@ -53,9 +52,9 @@ int main() {
     }
 
 #if CLIENT_TEST
```
Comment (kilograham) [inline pico_w/iperf/picow_iperf.c:54]: maybe we could rename this vestigial "_TEST" from when these were tests ;-)

**Comment by lurch** [comment]
I know nothing about FreeRTOS, but looking at the content of the diff in this PR, does `xstr` need removing from `picow_freertos_iperf.c` too? :shrug:

**Comment by peterharperuk** [comment]
> I know nothing about FreeRTOS, but looking at the content of the diff in this PR, does `xstr` need removing from `picow_freertos_iperf.c` too? 🤷

Oops - yes. I should test it really!

**Comment by lurch** [comment]
@kilograham / @liamfraser I wonder if it might be worth modifying https://github.com/raspberrypi/pico-examples/blob/master/.github/workflows/cmake.yml to do several different builds, with different combinations of `#define` s set? :man_shrugging:

**Comment by peterharperuk** [comment]
Added iperf client to freertos and checked they work.

**Comment by peterharperuk** [comment]
> @kilograham / @liamfraser I wonder if it might be worth modifying https://github.com/raspberrypi/pico-examples/blob/master/.github/workflows/cmake.yml to do several different builds, with different combinations of `#define` s set? 🤷‍♂️

Maybe this should be a separate issue? I don't really understand how the builds are configured but it looks a bit like pico_w examples are not built? To build everything...

-DPICO_BOARD=pico_w  -DWIFI_SSID='xxx' -DWIFI_PASSWORD='xxx' -DFREERTOS_KERNEL_PATH=/source/FreeRTOS-Kernel

**Comment by lurch** [comment]
> Maybe this should be a separate issue?

I'll let @kilograham decide if he wants to create an issue for it, since he knows _far_ more about the SDK (and how its builds work) than I do :wink:

**Comment by peterharperuk** [comment]
True - it should build for all boards

**Comment by peterharperuk** [comment]
I can't be bothered with this

**Pull request closed without merging**